op-node/rollup/derive/engine_queue.go

```diff
@@ -616,12 +522,11 @@ func (eq *EngineQueue) consolidateNextSafeAttributes(ctx context.Context) error
 	eq.ec.SetPendingSafeL2Head(ref)
 	if eq.safeAttributes.isLastInSpan {
 		eq.ec.SetSafeHead(ref)
-		eq.needForkchoiceUpdate = true
-		eq.postProcessSafeL2()
 	}
 	// unsafe head stays the same, we did not reorg the chain.
 	eq.safeAttributes = nil
 	eq.logSyncProgress("reconciled with L1")
+	eq.postProcessSafeL2()
```

Comment:
Seems suspicious that this moved out of the `isLastInSpan` if block.

Reply:
You're right - it pulls data from two different sources (the L2 safe head & the origin of the pipeline) so it's not safe to call in the middle of a span batch update.